hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/batch/DirectBufferUtils.java

```diff
@@ -0,0 +1,78 @@
+package pl.allegro.tech.hermes.consumers.consumer.batch;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+
+class DirectBufferUtils {
+    private static final Logger logger = LoggerFactory.getLogger(DirectBufferUtils.class);
+
+    private static final DirectBufferCleaner CLEANER;
+
+    static {
+        DirectBufferCleaner cleaner;
+        try {
+            cleaner = new Java11DirectBufferCleaner();
```

Comment:
What do you think about checking Java version at runtime, and based on that result try to initialize proper cleaner?

Reply:
done